### ISSUE TITLE
Improve S3927: Interfaces should be excluded

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6331]
### Description

Unexpected behavior for S3927 (ImplementSerializationMethodsCorrectly).

The suggestion on interfaces methods is to `make this method non-public`.

Interfaces methods that are decorated with the serialization attibutes are ignored, even if they are inherited by the implementation class.

A more proper behavior might be to exclude interfaces from the rule and maybe create another rule that warns the user to **not** use the `(De)serialization` attributes on interfaces.

### Repro steps

```csharp
using System.Runtime.Serialization.Formatters.Binary;
using System.Runtime.Serialization;

TestSimpleObject obj = new TestSimpleObject();

Stream stream = File.Open("DataFile.dat", FileMode.Create);
BinaryFormatter formatter = new BinaryFormatter();

try
{
#pragma warning disable SYSLIB0011
    formatter.Serialize(stream, obj);

    stream.Close();
    stream = File.Open("DataFile.dat", FileMode.Open);

    obj = (TestSimpleObject)formatter.Deserialize(stream);

}
finally
{
    stream.Close();
}

public interface ITest
{
    [OnSerializing]
    protected abstract void OnSerializingMethod(StreamingContext context); // Never called

    [OnSerialized]
    protected abstract void OnSerializedMethod(StreamingContext context); // Never called

    [OnDeserializing]
    protected void OnDeserializingMethod(StreamingContext context) // Never called
    {
        Console.WriteLine(nameof(OnDeserializingMethod));
    }

    [OnDeserialized]
    protected virtual void OnDeserializedMethod(StreamingContext context) // Never called
    {
        Console.WriteLine(nameof(OnDeserializedMethod));
    }
}

// This is the object that will be serialized and deserialized.
[Serializable]
public class TestSimpleObject : ITest
{
    public void OnSerializedMethod(StreamingContext context) // Never called
    {
        Console.WriteLine(nameof(OnSerializedMethod));
    }

    [OnSerializing]
    public void OnSerializingMethod(StreamingContext context) // Actually called, because it is re-decorated with [OnSerializing]
    {
        Console.WriteLine(nameof(OnSerializingMethod));
    }
}

```

**Comment by antonioaversa** [comment]
Stressing that "Re-decoration" is required for all serialization attributes, due to `Inherited=false` in the definition of each of the attributes, c.p.
https://learn.microsoft.com/en-us/dotnet/api/system.runtime.serialization.onserializingattribute?view=net-7.0

`Inherited=false` is what allows the rule to be safely disabled for interfaces, since any `public abstract` method in the interface, derived in a class implementing such an interface, **can** decide whether to re-declare the serialization attribute or not.
